lib/httparty/request.rb

```diff
@@ -276,7 +276,7 @@ def handle_response(body, &block)
           unless options[:maintain_method_across_redirects] && options[:resend_on_redirect]
             self.http_method = Net::HTTP::Get
           end
-        else
+        elsif last_response.code.to_s != '307' && last_response.code.to_s != '308'
```

Comment:
Maybe do to_i then instead. If they are integers it doesn't hurt anything and if they aren't, it seems more efficient than the strings. Minor, but everything adds up.